tsdb/series_index.go

```diff
@@ -155,8 +155,10 @@ func (idx *SeriesIndex) Delete(id uint64) {
 
 // IsDeleted returns true if series id has been deleted.
 func (idx *SeriesIndex) IsDeleted(id uint64) bool {
-	_, ok := idx.tombstones[id]
-	return ok
+	if _, ok := idx.tombstones[id]; ok {
+		return true
+	}
+	return idx.FindOffsetByID(id) == 0
 }
```

Comment:
Tombstones only exist for IDs since the last compaction so we now check if the ID exists in the map as well.